cylc/flow/workflow_files.py

```diff
@@ -649,31 +665,50 @@ def init_clean(
     """
     local_run_dir = Path(get_workflow_run_dir(reg))
     try:
-        _clean_check(reg, local_run_dir)
+        _clean_check(opts, reg, local_run_dir)
     except FileNotFoundError as exc:
         LOG.info(str(exc))
         return
 
+    # Check dir does not contain other workflows:
+    contained_workflows = asyncio.get_event_loop().run_until_complete(
+        get_contained_workflows(local_run_dir, MAX_SCAN_DEPTH + 1)
+    )  # Note: increased scan depth for safety
+    _suppress_no_db_msg = False
+    if len(contained_workflows) == 1:
+        init_clean(contained_workflows[0], opts)
+        _suppress_no_db_msg = True
+    elif len(contained_workflows) > 1:
+        bullet = "\n    - "
+        msg = (
+            f"{local_run_dir} contains the following workflows:"
+            f"{bullet}{bullet.join(contained_workflows)}"
+        )
+        if not opts.force:
+            raise WorkflowFilesError(f"Cannot clean - {msg}")
```

Comment:
Might be worth adding a note to the end of the message to the effect that the user can use `--force` if that is what they genuinely want to do...